src/Microsoft.Restier.Core/ApiBase.cs

```diff
@@ -67,7 +67,14 @@ protected ApiConfiguration ApiConfiguration
                         var builder = this.ConfigureApi(new ApiBuilder());
                         EnableConventions(builder, apiType);
                         ApiConfiguratorAttribute.ApplyApiBuilder(apiType, builder);
-                        builder.TryUseSharedApiScope(); // TODO: Maybe default to context scope?
+                        if (!builder.HasService<IApi>())
+                        {
+                            builder.Services
+                                .AddScoped(apiType, sp => sp.GetService<ApiScope>().Api)
```

Comment:
This is to ease those custom services that written only for a specific API type, so that they can GetApiService<MyApi>() and access its unique members directly, or even inject MyApi to a service's ctor. Otherwise they'd have to cast first.
I know this scenario is rare, maybe doesn't worth special care, let me remove them if you think unnecessary.